lib/golomb.js

```diff
@@ -346,8 +349,15 @@ function compare(a, b) {
 }
 
 function getPushes(items, script) {
+  try {
+    script.compile();
```

Comment:
Why is this necessary? This will hurt performance greatly.

Reply:
@chjj IIRC data pushes from some invalid scripts were still being added to the items to be indexes, causing a fork in the filter chain. I can dig into the exact block but OTOH it was around 10k height in testnet.

Reply:
as per the BIP: Each data push in the scriptPubKey of each output, *only if* the scriptPubKey is parseable

Reply:
@tuxcanfly, we can check for that a quicker way. The scripting system tracks bad pushdatas with `op.value === -1`. I'll merge this and add the check.